message/infrastructure/test_pubsub.go

```diff
@@ -640,6 +645,53 @@ func TestMessageCtx(t *testing.T, pubSub PubSub, features Features) {
 	}
 }
 
+func TestSubscribeCtx(t *testing.T, pubSub PubSub, features Features) {
```

Comment:
How about a test like this:

1) Create a subscriber
2) Create one subscription with ctx1
3) Create another sub with ctx2
4) After canceling ctx1, sub1 should be dead and sub2 should go on
5) ???
6) PROFIT

I think that this is possibly quite an important part of how ctx interacts with `Subscribe(ctx, topic)` and the end users are going to expect this behavior. 
However, some pubsubs would not pass it right away (see discussion of HTTP subscriber)